test/hotspot/jtreg/compiler/loopopts/TestReassociateInvariants.java

```diff
@@ -57,6 +64,12 @@ public int test1(int a, int b) {
         return v;
     }
 
+    // Range Check Elimination only happens once a + (b + i) is
+    // transformed into i + (a + b). With the range check eliminated,
+    // the loop can be removed. At this point, C2 doesn't support
+    // removal of long counted loop. The long counted loop is
+    // transformed into a loop nest with an inner int counted
+    // loop. That one is empty and is removed.
```

Comment:
Sounds like we should file an RFE for long counted loop removal, right?

Reply:
https://bugs.openjdk.org/browse/JDK-8369976